modules/@apostrophecms/attachment/index.js

```diff
@@ -694,11 +694,17 @@ module.exports = {
             // apos.attachment.url with the returned object
             for (let i = ancestors.length - 1; i >= 0; i--) {
               const ancestor = ancestors[i];
-              const fields = ancestor.imagesFields && ancestor.imagesFields[o._id];
-              if (fields) {
+              const ancestorFields = ancestor.attachment && ancestor.attachment._id === value._id && ancestor._fields;
+              if (ancestorFields) {
                 value = _.clone(value);
-                value._crop = _.pick(fields, 'top', 'left', 'width', 'height');
-                value._focalPoint = _.pick(fields, 'x', 'y');
+                o.attachment = value;
+                value._crop = {
+                  top: ancestorFields.top || 0,
+                  left: ancestorFields.left || 0,
+                  width: ancestorFields.width || 0,
+                  height: ancestorFields.height || 0
```

Comment:
For some reason, the image's `_field`  has its `left` property set to `null` , when setting 0 or nothing in the model for `left` _and_ `top` (which is very weird).

I spent too much time trying to find the source of the problem, so I dirty-fixed that by setting a default 0 value, in order to avoid having `null` in path...

```js
_fields: { top: 0, left: null, width: 50, height: 50, x: 83, y: 56 }
```

Reply:
Defaulting the width and height to 0 would not make sense, if we must default them then we should default them to the actual width and height of the image.

Reply:
I you guys have an idea of what could set `top` to `0` (which is good) and leave `left` to `null` in the image `_fields`, I'd love to have a hint :)

Reply:
> Defaulting the width and height to 0 would not make sense, if we must default them then we should default them to the actual width and height of the image.

yes that's what I thought also, I'll change that 👍

Reply:
@boutell I couldn't use the default attachment width and height because the file does simply not exist (when no cropping has been done, the attachment path does not contain the width nor height values).

I just left them as they are